lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -3204,16 +3205,102 @@ DefConstructor('\@@generic@caption[]{}', "<ltx:text class='ltx_caption'>#2</ltx:
+        my $block_box = $document->getNodeBox($block);
+        $block_width = $block_box->getWidth->valueOf(); }
+
+      if (@new_contents && ($current_width + $block_width > $textwidth)) {
+        # break when we exceed the line width, to keep close with the PDF arrangement
+        my $break = $document->insertElementBefore($block, 'break', 'class' => 'ltx_break');
+        push(@new_contents, $break);
+        $current_width = $block_width; }
+      else {
+        $current_width += $block_width; } }
+    push(@new_contents, $block); }
+  return @new_contents; }
+
+sub add_subfigure_row_sizes {
+  my ($document, $node, $whatsit, @contents) = @_;
+  my @row = ();
+  for my $child (@contents) {
+    if ($document->getNodeQName($child) eq 'ltx:break') {
+      next unless @row;
+      my $row_size      = scalar(@row);
+      my $row_size_attr = "ltx_flex_size_$row_size";
+      if ($row_size > 4) {    # add a generic class for larger rows, to style together easily
+        $row_size_attr .= " ltx_flex_size_many"; }
+      for my $cell (@row) {
+        my $class = $cell->getAttribute('class') || '';
+        $class .= ' ' if $class;
+        $cell->setAttribute('class', $class . $row_size_attr); }
+      @row = (); }
+    else {
+      push(@row, $child); } }
+  # Don't forget the last row, if any
+  if (@row) {
+    my $row_size      = scalar(@row);
+    my $row_size_attr = "ltx_flex_size_$row_size";
+    if ($row_size > 4) {    # add a generic class for larger rows, to style together easily
+      $row_size_attr .= " ltx_flex_size_many"; }
+    for my $cell (@row) {
+      my $class = $cell->getAttribute('class') || '';
+      $class .= ' ' if $class;
+      $cell->setAttribute('class', $class . $row_size_attr); } }
+  return; }
+
+# Feature 1: Add flex classes to each subfigure cell in a figure.
+# Feature 2: Add identifiers on each figure
+sub SubfigureAndID {
+  my ($document, $node, $whatsit, $id_kind) = @_;
+  # 1. Flex
+  # A little verbose, but much easier to do than XSLT 1.0 would allow.
+  my $contents_count = 0;    # count with author breaks excluded
+  my @contents       = grep { my $name = $document->getNodeQName($_);
+    my $ok = ($name =~ /^ltx:(?:graphics|float|figure|table|tabular|break|inline-para|inline-block|listing)$/);
+    $contents_count += 1 if $ok and $name ne 'ltx:break';
+    $ok; } element_nodes($node);
+  if ($contents_count > 1) {
+    @contents = add_subfigure_row_breaks($document, $node, $whatsit, @contents);
+    add_subfigure_row_sizes($document, $node, $whatsit, @contents); }
+  # 2. IDs
+  # Note that even without \caption, we'd probably like to have xml:id.
+  return GenerateID($document, $node, $whatsit, $id_kind); }
+
+Tag('ltx:figure', afterClose => sub { SubfigureAndID(@_, 'fig'); });
+Tag('ltx:float',  afterClose => sub { SubfigureAndID(@_, 'tab'); });
+Tag('ltx:table',  afterClose => sub { SubfigureAndID(@_, 'tab'); });
```

Comment:
Just for the record: Tag accumulates the daemons, so you could have just added your subfigure code with separate `Tag`s.